### PR TITLE
Config adjustments to handle blank enironment variables

### PR DESCRIPTION
**Pull request by mmulich**
## Pull Request Overview

- [Use the default config variable when env-var is empty](https://github.com/WildMeOrg/houston/commit/bfe3e0d81782148ce81c3425e6f8c9e889ab7f10) -- This was added to allow for the null definition of `MAIL_DEFAULT_SENDER_EMAIL`. This is a generally useful addition for handling the case where the environment variable is defined, but only contains an empty string.
- [DRY the sentry config and use a single variable name](https://github.com/WildMeOrg/houston/commit/a94bfa3f128bfa5979c251316ef174de2b0ac796) -- I feel centralizing the setting makes sense; thus DRY the code. I can see the desire to distinguish between development vs production, especially when you have a default for development, but I'm going to opt for non-default telemetry. Development errors are going to be many; so providing a default means lots of noise in sentry that will result in no one using sentry.

---

Requires a change in deployed configurations, where the `SENTRY_DSN_PRODUCTION` environment variable should be renamed to `SENTRY_DSN`.

### CODE REVIEW

**Review by karenc — APPROVED** [review]

File: config/base.py [hunk 1]
```diff
@@ -14,6 +14,15 @@
 DATA_ROOT = Path(os.getenv('DATA_ROOT', HERE / '_db'))
 
 
+def _getenv(var_name, default):
```
Comment (bluemellophone) [inline config/base.py:17]: This function is a nice simple wrapper around `os.getenv(variable_name)`, but how does it differ functionally from using simply `os.getenv(variable_name, default_value)`?  If we are going to use this function, let's add a `default=None` to this function's signature and upgrade _all_ instances within `config/base.py` to use `_getenv`.  We could also likely upgrade the usage in a handful of config / boilerplate files in the project:

- `tasks/__init__.py`
- `integration_tests/conftest.py`
- `config/codex.py`
- `config/__init__.py`
- `config/mws.py`
- `config/utils.py`
- `config/base.py`
- `tests/conftest.py`
- `scripts/docs/generate_models_graph.py`

**Review by bluemellophone — CHANGES_REQUESTED** [review]
LGTM, small comment / request for change on standardizing the parsing of our envars.

File: config/base.py (hunk 1 above)
Comment (karenc) [reply config/base.py:17]: The difference is with `os.getenv(variable_name, default_value)`, the `default_value` is only used if `variable_name` is not in the environment, but `_getenv(variable_name, default_value)` returns `default_value` if `variable_name` is not in the environment or if `variable_name` is empty.  If we rely on being able to set an environment variable to an empty string then we can't use `_getenv`

Comment (karenc) [reply config/base.py:17]: This is equivalent to what `os.getenv` is doing:

```python
if variable_name in os.environ:
    return os.environ[variable_name]
return default_value
```

This is equivalent to what `_getenv` is doing:

```python
if os.environ.get(variable_name):
    return os.environ[variable_name]
return default_value
```

So for example, if I have a environment variable `RECAPTCHA_PUBLIC_KEY` and it's an empty string, `os.getenv('RECAPTCHA_PUBLIC_KEY', 'xxx')` returns `""`, `_getenv('RECAPTCHA_PUBLIC_KEY', 'xxx')` returns `"xxx"`.

Are you suggesting we add a flag to `_getenv` to support both cases?

Comment (bluemellophone) [reply config/base.py:17]: Yes, I think it is handy to make it clear which envars allow us to accept empty fields

**Review by bluemellophone — APPROVED** [review]
LGTM

File: config/base.py (hunk 1 above)
Comment (bluemellophone) [reply config/base.py]: Thank you for making this change.  I very much prefer the explicit flag for `empty_ok`.

Comment (karenc) [reply config/base.py]: :+1: I think it's good too

**Pull request merged**